chia/plotting/inventory_plots.py

```diff
@@ -0,0 +1,52 @@
+import logging
+from collections import Counter
+from pathlib import Path
+from typing import Dict, List
+
+from blspy import G1Element
+from chiapos import Verifier
+
+from chia.plotting.plot_tools import find_duplicate_plot_IDs, get_plot_filenames, load_plots, parse_plot_info
+from chia.util.config import load_config
+from chia.util.hash import std_hash
+from chia.util.keychain import Keychain
+from chia.wallet.derive_keys import master_sk_to_farmer_sk, master_sk_to_local_sk
+
+log = logging.getLogger(__name__)
+
+
+def inventory_plots(root_path, grep_string, plot_public_key):
+    config = load_config(root_path, "config.yaml")
+
+    v = Verifier()
+    log.info("Loading plots in config.yaml using plot_tools loading code\n")
+    kc: Keychain = Keychain()
+    pks = [master_sk_to_farmer_sk(sk).get_g1() for sk, _ in kc.get_all_private_keys()]
+    pool_public_keys = [G1Element.from_bytes(bytes.fromhex(pk)) for pk in config["farmer"]["pool_public_keys"]]
+    _, provers, failed_to_open_filenames, no_key_filenames = load_plots(
+        {},
+        {},
+        pks,
+        pool_public_keys,
+        grep_string,
+        False,
+        root_path,
+        open_no_key_filenames=True,
+    )
+
+    searching_for_plot_public_key = False
+
+    if plot_public_key is not None:
+        log.info(f"only looking for plot with a public key of {plot_public_key}")
+        searching_for_plot_public_key = True
+
+
+    for plot_path, plot_info in provers.items():
+        pr = plot_info.prover
+
+        if searching_for_plot_public_key:
+            plot_public_key_matches = plot_public_key in str(plot_info.plot_public_key)
+
+        if not searching_for_plot_public_key or plot_public_key_matches:
```

Comment:
If `searching_for_plot_public_key` is true, then `plot_public_key_matches` is always `plot_public_key in str(plot_info.plot_public_key)`. If it's false then this will always be true. As such, it seems the above lines 47 and 48 are redundant and you can just go 
```python
if not searching_for_plot_public_key or plot_public_key in str(plot_info.plot_public_key):
```
In fact I think the mutable variable `searching_for_plot_public_key` is generally redundant and can be replaced with `plot_public_key` itself. So this clause here would read
```python
if (plot_public_key is None) or plot_public_key in str(plot_info.plot_public_key):
```
Would this spare 4 lines and two variables?

Reply:
Thanks @Nikolaj-K I've taken you up on this and pushed the changes.